### PR TITLE
Move cache to state instead of contest

### PR DESCRIPTION
**Pull request by rpiaggio**
Happens to also fix the issue of blank explore when navigating to the root.

### CODE REVIEW

**Comment by rpiaggio** [comment]
Subsumes #2856 

Also, reinitializes cache update fibers upon program change.

File: explore/src/main/scala/explore/programs/ProgramsPopup.scala [hunk 1]
```diff
@@ -53,6 +53,7 @@ object ProgramsPopup {
     .useContext(AppContext.ctx)
     .useStateView(IsOpen(true))
     .render { (props, ctx, isOpen) =>
+      println("hello")
```
Comment (hugo-vrijswijk) [inline explore/src/main/scala/explore/programs/ProgramsPopup.scala:56]: Accidentally added?

File: common/src/main/scala/explore/cache/CacheComponent.scala [hunk 2]
```diff
@@ -6,37 +6,36 @@ package explore.cache
...
 import japgolly.scalajs.react.*
 import japgolly.scalajs.react.feature.Context
 import japgolly.scalajs.react.util.DefaultEffects.{Async => DefaultA}
+import japgolly.scalajs.react.util.DefaultEffects.{Sync => DefaultS}
 import japgolly.scalajs.react.vdom.html_<^.*
 import lucuma.ui.syntax.pot.*
 
-trait CacheComponent[Props: Reusability, A]:
-  private type F[T] = DefaultA[T]
+import scala.tools.nsc.doc.model.Def
```
Comment (toddburnside) [inline common/src/main/scala/explore/cache/CacheComponent.scala:20]: Is this being used? Or was metals being helpful?

**Review by toddburnside — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/programs/ProgramsPopup.scala (hunk 1 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/programs/ProgramsPopup.scala:56]: Good catch :D

File: common/src/main/scala/explore/cache/CacheComponent.scala (hunk 2 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/cache/CacheComponent.scala:20]: Metals, definitely

**Pull request merged**